adapters/gumgum/gumgum.go

```diff
@@ -0,0 +1,135 @@
+package gumgum
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type GumGumAdapter struct {
+	URI string
+}
+
+func (g *GumGumAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	var validImps []openrtb.Imp
+	var trackingId string
+
+	numRequests := len(request.Imp)
+	errs := make([]error, 0, numRequests)
+
+	for i := 0; i < numRequests; i++ {
+		imp := request.Imp[i]
+		zone, err := preprocess(&imp)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			if request.Imp[i].Banner != nil {
+				bannerCopy := *request.Imp[i].Banner
+				if bannerCopy.W == nil && bannerCopy.H == nil && len(bannerCopy.Format) > 0 {
+					format := bannerCopy.Format[0]
+					bannerCopy.W = &(format.W)
+					bannerCopy.H = &(format.H)
+				}
+				request.Imp[i].Banner = &bannerCopy
+				validImps = append(validImps, request.Imp[i])
+				trackingId = zone
+			}
+		}
+	}
+
+	if len(validImps) == 0 {
+		return nil, errs
+	}
+
+	request.Imp = validImps
+
+	if request.Site != nil {
```

Comment:
I notice in your `static/bidder-info` that you also support apps... did you want to do something with the tracking ID for those here too? Maybe in `req.App`?

It seems a bit weird that the `zone` info doesn't make it anywhere into the request that goes to your servers on App requests.

Reply:
Sorry that's an oversight on my end. GumGum will only support site for now so I'll correct our bidder-info file. Thanks for catching that! If I modify that file will you have to reapprove?

Reply:
yeah, but it's fine. We're active :)